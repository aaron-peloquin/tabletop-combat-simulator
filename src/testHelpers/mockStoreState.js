export default {
  "creatures": [
    {
      "name": "Goplin",
      "cr": "1/4",
      "hash": "hivn3p0k",
      "description": "A stout monster",
      "defaultInitiative": 10,
      "armor": 11,
      "hp": 5,
      "hitDiceEquation": "1d20+1",
      "damageDiceEquation": "1d4"
    },
    {
      "name": "Elven Ranger",
      "cr": "1/2",
      "hash": "h3dxf5wl",
      "description": "A swift hunter",
      "defaultInitiative": 16,
      "armor": 13,
      "hp": 10,
      "hitDiceEquation": "1d20+8",
      "damageDiceEquation": "1d6+3"
    },
    {
      "name": "Knight",
      "cr": "1",
      "hash": "hee9p286",
      "description": "An honorable soldier",
      "defaultInitiative": 11,
      "armor": 18,
      "hp": 25,
      "hitDiceEquation": "1d20+4",
      "damageDiceEquation": "1d12+4"
    },
    {
      "name": "Lizardfolk",
      "cr": "3",
      "hash": "howx26rt",
      "description": "An ancient sneaky monster",
      "defaultInitiative": 14,
      "armor": 14,
      "hp": 15,
      "hitDiceEquation": "1d20+5",
      "damageDiceEquation": "1d10+2"
    }
  ],
  "sideBar": false,
}