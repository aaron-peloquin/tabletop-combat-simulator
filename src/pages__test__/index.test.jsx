// import renderer from "react-test-renderer"
import { connect } from "react-redux"
import { shallowWithStore } from "enzyme-redux"
import { createMockStore } from "redux-test-utils"
import mockStoreState from "../testHelpers/mockStoreState"
import toJson from "enzyme-to-json"

import Index from "../pages/index"

describe("<Index /> page", ()=>{
  it("loads", async () => {
    const ReactComponent = () => <Index />
    expect(typeof ReactComponent).toBe("function")
  })

  it("Snapshots", () => {
    const ReactComponent = () => <Index />
    const mapStateToProps = (state) => ({state})
    const store = createMockStore(mockStoreState)
    const ConnectedComponent = connect(mapStateToProps)(ReactComponent)
    const component = shallowWithStore(<ConnectedComponent />, store)
    expect(toJson(component)).toMatchSnapshot()
  })
})