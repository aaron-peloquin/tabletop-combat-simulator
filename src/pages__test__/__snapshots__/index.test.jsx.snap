// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Index /> Page Snapshots 1`] = `
<WithStyles(Grid)
  alignItems="stretch"
  className="Index-CenterItems-1"
  container={true}
  justify="center"
  spacing={24}
>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <Connect(WithStyles(CreatureForm)) />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    container={true}
    item={true}
    justify="center"
    spacing={24}
    xs={12}
  >
    <WithStyles(Grid)
      item={true}
      md={6}
    >
      <Connect(TeamList)
        Team="a"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      md={6}
    >
      <Connect(TeamList)
        Team="b"
      />
    </WithStyles(Grid)>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <Connect(SimulateCombat) />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    container={true}
    item={true}
    justify="center"
    spacing={24}
    xs={12}
  >
    <WithStyles(Grid)
      item={true}
      md={6}
    >
      <WithStyles(Connect(SimulationResults)) />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      md={6}
    >
      <WithStyles(Connect(SimulationLog)) />
    </WithStyles(Grid)>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <Connect(DeleteAllCreatures) />
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
